src/OpenTracingActiveRecordBehavior.php

```diff
@@ -0,0 +1,165 @@
+     * @param CEvent $event event parameter
+     */
+    public function afterSave($event)
+    {
+        if(!$this->traceSave && $this->activeScope) {
+            return;
+        }
+        $this->activeScope->getSpan()
+            ->setTag('db.active_record.primary_key', json_encode($this->getOwner()->getPrimaryKey()));
+        $this->activeScope->close();
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onBeforeDelete} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * You may set {@link CModelEvent::isValid} to be false to quit the deletion process.
+     * @param CEvent $event event parameter
+     */
+    public function beforeDelete($event)
+    {
+        if(!$this->traceDelete && \Yii::app()->hasComponent($this->opentracing)) {
+            return;
+        }
+
+        $options = $this->spanOptionsFromActiveRecordClass();
+        $options['tags']['db.active_record.primary_key'] = json_encode($this->getOwner()->getPrimaryKey());
+
+        $this->activeScope = \Yii::app()->hasComponent($this->opentracing)
+            ->startActiveSpan($this->spanName('DELETE'), $options);
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onAfterDelete} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function afterDelete($event)
+    {
+        if(!$this->traceDelete && !$this->activeScope) {
+            return;
+        }
+        $this->activeScope->close();
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onBeforeFind} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function beforeFind($event)
+    {
+        if(!$this->traceFind && !\Yii::app()->hasComponent($this->opentracing)) {
+            return;
+        }
+
+        $this->activeScope = \Yii::app()->hasComponent($this->opentracing)
+            ->startActiveSpan($this->spanName('FIND'), $this->spanOptionsFromActiveRecordClass());
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onAfterFind} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function afterFind($event)
+    {
+        if(!$this->traceFind && !$this->activeScope) {
+            return;
+        }
+
+        $this->activeScope->getSpan()->log(['criteria' => $this->getOwner()->getDbCriteria()->toArray()]);
+        $this->activeScope->close();
+    }
+
+    private function spanName($action)
+    {
+        return sprintf('DB-AR::%s %s', strtoupper($action), get_class($this->getOwner()));
```

Comment:
this kind of span name is too specific, please make it more general

> An operation name, a human-readable string which concisely represents the work done by the Span (for example, an RPC method name, a function name, or the name of a subtask or stage within a larger computation). The operation name should be the most general string that identifies a (statistically) interesting class of Span instances. That is, "get_user" is better than "get_user/314159".

Reply:
Operation name is for example: `DB-AR::UPDATE \activerecord\classname`, this string just identifies, that given class is performing some action. There is no id. Should i make it even less specific?